rustls/src/crypto/aws_lc_rs/sign.rs

```diff
@@ -117,7 +117,7 @@ impl RsaSigningKey {
             }
         }
         .map_err(|key_rejected| {
-            Error::General(format!("failed to parse RSA private key: {}", key_rejected))
+            Error::General(format!("failed to parse RSA private key: {key_rejected}"))
```

Comment:
I think this commit comes from  `cargo clippy --fix`, but I think `cargo clippy --fix --all-features --all-targets` produces a more uniform result. eg, this link also fires on `src/crypto/ring/sign.rs`

Reply:
> I think this commit comes from cargo clippy --fix, but I think cargo clippy --fix --all-features --all-targets produces a more uniform result. 

Ahh yes, you're right. Let me rework.

Reply:
I prefer your commit message, but here's the same commit I came up with https://github.com/rustls/rustls/commit/9b0c22d94b82524d1d4723f94afc5123fbafa0ca

Reply:
I regenerated that commit (and while reviewing the diff, also noticed a small manual cleanup opportunity I tacked on separate from the machine generated diff: 4d4a01436e40ad62ab54d37646ae1bbb9c803508)

Reply:
Ohhh right, that was an intentional thing because of `no-std`: 

```
error: unused variable: `invalid`
   --> rustls/src/msgs/handshake.rs:311:44
    |
311 |                 HostNamePayload::IpAddress(invalid) => {
    |                                            ^^^^^^^ help: if this is intentional, prefix it with an underscore: `_invalid`
    |
    = note: `-D unused-variables` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(unused_variables)]`
````

Will drop :sweat:

Reply:
shower thought: i wonder if there is a way to avoid this by the logging macros becoming `drop(format!(<args>))` or something? But obviously we'd need to be confident that the compiler doesn't actually do the formatting...